src/mol-plugin-ui/structure/selection.tsx

```diff
@@ -219,6 +219,11 @@ export class StructureSelectionActionsControls extends PluginUIComponent<{}, Str
 
     render() {
         const granularity = this.plugin.managers.interactivity.props.granularity;
+        const granularitySelect = StructureSelectionParams.granularity;
+        const granularityOptions = this.plugin.spec.components?.selectionTools?.granularityOptions;
+        if (granularityOptions) {
+            granularitySelect.options = granularitySelect.options.filter(([firstItem]) => granularityOptions.has(firstItem));
+        }
```

Comment:
- This will never change, so you can just compute this once say in a constructor.
- Also, do not overwrite the global `StructureSelectionParams`, but instead create a copy with `{ ...StructureSelectionParams, options: ... }`